pkg/url/url.go

```diff
@@ -86,24 +86,40 @@ func Get(client *occlient.Client, localConfig *config.LocalConfigInfo, urlName s
 }
 
 // GetIngress returns ingress spec for given URL name
-func GetIngress(kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, urlName string) (iextensionsv1.Ingress, error) {
-
+func GetIngress(kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, urlName string, componentName string) (URL, error) {
+	remoteExist := true
 	// Check whether remote already created the ingress
 	ingress, err := kClient.GetIngress(urlName)
-	if err == nil {
-		return *ingress, nil
+	if err != nil {
+		remoteExist = false
```

Comment:
I think this should return the error and exit immediately.

Reply:
We should not exit with an error if the ingress does not exist on cluster. We need to check if the ingress exist locally as well, to determine the URL state.

Reply:
But in case of an connection error, it's better to inform the user that there was a problem with the network and thus the error. Otherwise he might be surprised due to inconsistency between the cluster and the local.

Reply:
If the ingress does not exist on cluster, there will also see this error. It's not only due to connection. If a user has a URL created locally and not pushed to cluster, this error will be returned when we try to get ingress from cluster. However, we should show user the URL state as `Not Pushed`, and inform user there is local change, they need to do a push to get the change reflected. 
```
$ odo url list
Found the following URLs for component java-spring-boot
NAME            STATE          URL                                 PORT     SECURE
ingressurl1     Not Pushed     https://ingressurl1.1.2.3.4.com     8080     true

There are local changes. Please run 'odo push'.
```

Reply:
This is also how odo1 handles `url describe` for routes as well. 
https://github.com/openshift/odo/blob/20491afbba8c1a587a6cbbada204cbe3a12dd51b/pkg/url/url.go#L57-L61

Reply:
> If a user has a URL created locally and not pushed to cluster, this error will be returned when we try to get the ingress name from cluster.

Yes we can use Kerrors package and check if the cause is the absence of the ingress on the cluster. For reference https://github.com/openshift/odo/blob/20491afbba8c1a587a6cbbada204cbe3a12dd51b/pkg/occlient/occlient.go#L3330

> However, we should show user the URL state as Not Pushed, and inform user there is local change, they need to do a push to get the change reflected.

In case of network error, just inform the user and exit immediately otherwise he might be confused and won't have any idea about the underlying cause.

> This is also how odo1 handles url describe for routes as well.

That needs to be fixed too.

Reply:
Thanks Mrinal for the suggestion. I have updated the error check and tested the `Not Pushed` case for `url describe` still works as expected

Reply:
:+1: